src/bitops.c

```diff
@@ -16,12 +16,23 @@
 /* Count number of bits set in the binary array pointed by 's' and long
  * 'count' bytes. The implementation of this function is required to
  * work with an input string length up to 512 MB or more (server.proto_max_bulk_len) */
-__attribute__((target("popcnt")))
+#if defined(__x86_64__) && ((defined(__GNUC__) && __GNUC__ > 5) || (defined(__clang__)))
+    #if __has_attribute(target)
```

Comment:
I don't remember how old `__has_attribute` is but you may need to check `#if defined(__has_attribute)` 

e.g.
https://github.com/redis/redis/blob/3788a055fee9041860d422d2d93e42677552bceb/src/config.h#L145

Reply:
`__has_attribute` was introduced since gcc 5, but i don't mind to add it.

Reply:
done with https://github.com/redis/redis/pull/13609/commits/601be8000803f5d3870bffc8bc030bc583b8f2ae